third-party/realdds/src/dds-device-impl.cpp

```diff
@@ -110,6 +113,22 @@ void dds_device::impl::run( size_t message_timeout_ms )
             }
         }
     } );
+
+    if( _metadata_reader )  // Might not be present
+    {
+        _metadata_reader->on_data_available( [&]() {
+            topics::flexible_msg notification;
+            eprosima::fastdds::dds::SampleInfo info;
+            while( topics::flexible_msg::take_next( *_metadata_reader, &notification, &info ) )
+            {
+                if( ! notification.is_valid() )
+                    continue;
+
+                if( _on_metadata_available )
```

Comment:
Any chance this will change during runtime?
Up until now was to set the callbacks *before* run() is called. I propose that we keep to that and only set the callback if we have a callback.

Reply:
The callback is set during `dds_device_proxy` constructor and does not change during runtime.
Still need to check validity before dereferencing.

Reply:
But that's what I'm saying: if the callback isn't set when we `run()` then no need to set up another `on_data_available` callback...

Reply:
Currently `run` is being called from [on_device_added](https://github.com/IntelRealSense/librealsense/blob/a946b4babc46ac313f8dfd277886b33a026bb7aa/src/context.cpp#L1409) callback.
The callback is only set later in `dds_device_proxy` [constructor](https://github.com/IntelRealSense/librealsense/blob/a946b4babc46ac313f8dfd277886b33a026bb7aa/src/context.cpp#L1164).
@maloel Do you mean to move the call to `run` into `dds_device_proxy` constructor? And then I will only check if it is set once when I [check `_metadata_reader`](https://github.com/IntelRealSense/librealsense/pull/11412/files/73ff6001aa9b3721e7690dc490dcad46156a15c5#diff-83f09ab46adb800f881b6a5240dc38a6f64207359f8b8b33167134d82195a01aR117) existence?

Reply:
If so then we need to pass `message_timeout_ms` parameter to the dds_device_proxy